### PR TITLE
Remove previous interaction listeners if `interactions` prop did update

### PR DESCRIPTION
**Pull request by dtrucs**

### CODE REVIEW

**Review by CFarcy — APPROVED** [review]

File: src/modules/Map/services/mapUtils.js [hunk 1]
```diff
@@ -181,20 +185,23 @@ export function setInteractions ({ map, interactions, callback }) {
 
     const { interactions: filteredInteractionsSpec, feature, layerId } = interactionsSpec;
     filteredInteractionsSpec.forEach(interaction =>
-      callback({ event: e, map, layerId, feature, interaction, eventType }));
+      callback({ event, map, layerId, feature, interaction, eventType }));
   };
+
+  const debounceListener = debounce(
```
Comment (jrmi) [inline src/modules/Map/services/mapUtils.js:191]: Why do you add a debounce here ?

File: src/modules/Map/services/mapUtils.test.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ import {
   PREV_STATE,
 } from './mapUtils';
 
+jest.mock('lodash.debounce', () => jest.fn(fn => (...args) => fn(...args)));
```
Comment (jrmi) [inline src/modules/Map/services/mapUtils.test.js:13]: Did you add a test to verify event handler are actually removed then re-added ?

File: src/modules/Map/services/mapUtils.js (hunk 1 above)
Comment (dtrucs) [reply src/modules/Map/services/mapUtils.js:191]: I add it to replace the hacky debouncer with setInterval/clearInterval

File: src/modules/Map/services/mapUtils.test.js (hunk 2 above)
Comment (dtrucs) [reply src/modules/Map/services/mapUtils.test.js:13]: I just add some tests related to it 👍

**Review by jrmi — APPROVED** [review]

**Pull request merged**